### PR TITLE
fix(backend): restore fade subtitles and clip transitions

### PR DESCRIPTION
**Pull request by FujiwaraChoki**
## Summary
- replace invalid `with_effects()` lambda usage in fade subtitle backgrounds with MoviePy `CrossFadeIn`/`CrossFadeOut` effect objects
- replace invalid string-based transition effects with MoviePy `FadeIn`/`FadeOut` effect objects
- add backend regression tests covering both fade subtitle and transition effect application

## Why
Production logs for task `f802e7e0-c410-4837-b864-3b7cf702d764` on Hetzner showed MoviePy v2 raising `'function' object has no attribute 'copy'` for subtitle fades and `'str' object has no attribute 'copy'` for transitions. The worker fell back gracefully, but output quality degraded because fades and transitions were skipped.

## Verification
- `backend/.venv/bin/pytest --no-cov backend/tests/unit/test_video_utils_effects.py backend/tests/test_video_utils_diarization.py`
- confirmed the production worker logs on Hetzner match the reported failures before applying the fix

### CODE REVIEW

File: backend/src/video_utils.py [hunk 1]
```diff
@@ -1412,10 +1410,10 @@ def apply_transition_effect(
         fade_duration = 0.5  # Half second fade
 
         # Fade out clip1
-        clip1_faded = clip1.with_effects(["fadeout", fade_duration])
+        clip1_faded = clip1.with_effects([FadeOut(fade_duration)])
 
         # Fade in clip2
-        clip2_faded = clip2.with_effects(["fadein", fade_duration])
+        clip2_faded = clip2.with_effects([FadeIn(fade_duration)])
```
[the remark this answers is not in the document]
Comment (FujiwaraChoki) [reply backend/src/video_utils.py]: Fixed in dbea9a2. `apply_transition_effect()` now keeps the exported file aligned to clip 2 by compositing only the previous clip tail into the transition window and then appending the remainder of clip 2. That preserves the current clip duration and metadata instead of prepending the full previous segment. I also expanded the regression test to assert the exact subclip boundaries and verify we only concatenate the transition intro plus the remainder of clip 2.

**Comment by FujiwaraChoki** [comment]
Follow-up after checking the MoviePy docs: standalone SupoClip exports now intentionally avoid inter-clip transitions. The backend wrapper still exists for compatibility, but it delegates to plain clip generation so each exported clip keeps its own exact segment boundaries and metadata. Added regression coverage for both the wrapper behavior and the task-service path. True cross-clip transitions make more sense as a future stitched-reel mode than as part of independent clip generation.
